repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/keystonev2/selfvalidation/KeystoneV2SelfValidationTest.groovy

```diff
@@ -79,20 +75,7 @@ class KeystoneV2SelfValidationTest extends ReposeValveTest {
         mc.getHandlings().get(0).getRequest().getHeaders().getFirstValue("x-tenant-name") == "mytenantname"
 
         and: "The X-Authenticated-By headers should be present at the origin service."
-        def foundPassword = false
-        def foundMultiFactor = false
-        for (Header header : mc.getHandlings().get(0).getRequest().getHeaders()) {
-            if ("X-Authenticated-By".equalsIgnoreCase(header.name)) {
-                if ("Password".equalsIgnoreCase(header.value)) {
-                    foundPassword = true
-                }
-                if ("Mfa".equalsIgnoreCase(header.value)) {
-                    foundMultiFactor = true
-                }
-            }
-        }
-        assertTrue(foundPassword)
-        assertTrue(foundMultiFactor)
+        mc.getHandlings().get(0).getRequest().getHeaders().findAll("X-Authenticated-By").sort() == ["MFA", "PASSWORD"]
```

Comment:
LOL.

Yeah, it's all syntactic sugar.  Although to be clear, `handler.request` will call `handler.getRequest()`, not access the field directly.  To access the field, you would call `handler.@request`.  Also, I totally didn't know about "private" being meaningless in Groovy.  No wonder it lost the language war. 😔